packages/office-ui-fabric-react/src/components/ThemeGenerator/ThemeGenerator.ts

```diff
@@ -80,23 +80,22 @@ export class ThemeGenerator {
 
   /*
    * Gets code-formatted load theme blob that can be copy and pasted.
+   * Only used for the old theme designer, where loadTheme usage is acceptable,
+   * unlike in the new theme designer.
    */
   public static getThemeAsCode(slotRules: IThemeRules): any {
-    const attributeTemplate = "    {0}: '{1}',\n";
-    let output = '';
-
-    output += 'loadTheme({\n  palette: {\n';
+    const output = 'loadTheme({\n  palette: {\n';
+    return ThemeGenerator._makeRemainingCode(output, slotRules);
+  }
 
-    for (const ruleName in slotRules) {
-      if (slotRules.hasOwnProperty(ruleName)) {
-        const rule: IThemeSlotRule = slotRules[ruleName];
-        const camelCasedName = rule.name.charAt(0).toLowerCase() + rule.name.slice(1);
-        const outputColor = rule.color ? '#' + rule.color.hex : rule.value || '';
-        output += format(attributeTemplate, camelCasedName, outputColor);
-      }
-    }
-    output += '  }\n});';
-    return output;
+  /**
+   * Gets code-formatted load theme blob, specifically for the new theme designer,
+   * aka.ms/themedesigner. Can't use loadtheme like the old theme designer.
+   * We want to use the theme object from createTheme and use the Customizations.applySettings API instead.
+   */
+  public static getThemeAsCodeForThemeDesigner(slotRules: IThemeRules): any {
```

Comment:
One other comment, I wonder if it would be better to give this a name less specific to theme designer? Like `getCreateThemeCode` or `getThemeAsCodeWithCreateTheme` maybe. (I'd actually have preferred adding an extra prop `useCreateTheme` or something like that to the existing `getThemeAsCode`, contrary to JD's feedback. :P ) But it's not a big deal since I don't think this is a public API.

Reply:
Done.